src/melee.cpp

```diff
@@ -539,7 +539,8 @@ bool Character::melee_attack_abstract( Creature &t, bool allow_special,
         add_msg( m_bad, _( "This weapon is too unwieldy to attack with!" ) );
         return false;
     }
-    int move_cost = attack_speed( *cur_weapon );
+    int move_cost = attack_speed( cur_weapon->has_flag( flag_UNARMED_WEAPON ) ? *
+                                  ( new item() ) : *cur_weapon );
```

Comment:
In C++, new dynamically allocates an object, and returns a pointer to it.
Pointers returned by new must be manually `deleted` to prevent memory leaks.
To do this without memory leaks, all you'd need to do is remove the `new`.

However, this is probably more correct:
```suggestion
    int move_cost = attack_speed( cur_weapon->has_flag( flag_UNARMED_WEAPON ) ? 
                                  ( null_item_reference() ) : *cur_weapon );
```